### PR TITLE
New: Support TypeScript 2.4 (fixes #321)

### LINKED ISSUE #321: Support TypeScript 2.4
Main changes to syntax are dynamic imports and string enums. Both should probably "just work" by upgrading the `"typescript"` package.

https://blogs.msdn.microsoft.com/typescript/2017/06/12/announcing-typescript-2-4-rc/

### PR DESCRIPTION
**Pull request by azz**
Makes sense to keep support for 2.3 until 2.4 is stable upstream.

Fixes #321.

### CODE REVIEW

**Comment by azz** [comment]
Added a test for `import().then()` including replacing `TSImportKeyword` with `Import` as per [estree experimental/import-expression](https://github.com/estree/estree/blob/master/experimental/import-expression.md#import).

**Comment by JamesHenry** [comment]
@azz I have created a branch called `ts-2.4`, please can you reopen the PR with that as the base?

**Comment by azz** [comment]
Done

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "npm-license": "0.3.3",
     "shelljs": "0.7.7",
     "shelljs-nodecli": "0.1.1",
-    "typescript": "~2.3.2"
+    "typescript": "~2.4 || ~2.3.2"
```
Comment (JamesHenry) [inline package.json]: Thanks, as per my comments on the issue I feel this should still just be one version for now. We can potentially look to change this philosophy before `ts-2.4` is merged into master

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ A parser that converts TypeScript into an [ESTree](https://github.com/estree/est
 
 We will always endeavor to support the latest stable version of TypeScript.
 
-The version of TypeScript currently supported by this parser is `~2.3.2`. This is reflected in the `devDependency` requirement within the package.json file, and it is what the tests will be run against. We have an open `peerDependency` requirement in order to allow for experimentation on newer/beta versions of TypeScript.
+The version of TypeScript currently supported by this parser is `~2.3.2 || ~2.4`. This is reflected in the `devDependency` requirement within the package.json file, and it is what the tests will be run against. We have an open `peerDependency` requirement in order to allow for experimentation on newer/beta versions of TypeScript.
```
Comment (JamesHenry) [inline README.md]: See version comment

**Comment by soda0289** [comment]
I'm ok with having a range of versions but it is often not possible to do so. This release might be one where no breaking changes occur in the AST and its possible to have a version range. I like the idea of having a separate branch for now just to ensure no surprises come up.

**Comment by JamesHenry** [comment]
Thanks, @azz!

**Pull request merged**